azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest.java

```diff
@@ -361,10 +362,20 @@ private void assertAttempts(String name, int attempt) {
   
   private ExecutableFlow prepareExecDir(File execDir, String flowName,
       int execId) throws IOException {
-    synchronized (this) {
-      // clean interrupted status
-      Thread.interrupted();
-      FileUtils.copyDirectory(execDir, workingDir);
+    // copyDirectory has been failing randomly with ClosedByInterruptException. have some retries.
```

Comment:
Actually good question and I think that's because some tests still run `runner.run()` in the main unit test thread. Part of FlowRunner design is calling `interrupt()` on the FlowRunner thread to wake it up for work. Should be possible to get rid of this error by always running the FlowRunner in its own thread. I'll try switching to that and then can remove all these hacks from the test (including call to `Thread.interrupted();`).